### ISSUE TITLE
Vue state does not sync with kendo datasource

### ISSUE THREAD

**Issue by icqbox** [vuejs/vue#11469]
### Version
2.6.11

### Reproduction link
[http://dojo.telerik.com/aGENIHuW](http://dojo.telerik.com/aGENIHuW)

### Steps to reproduce
1. Click 'Test'
2. Click 'Update'

### What is expected?
Replace first row with second row of grid.

### What is actually happening?
No change on screen.

---
If I firstly click 'Test' (just only a same value assignment to state) and then click 'Update', strangely it does not work so the grid has no change. I found the Vue state was updated but the datasource won't.
If I firstly click 'Update', it works!

The problem is why, after clicked 'Test', the kendo grid datasource became cached and out of sync with the Vue state? If I don’t click 'Test', they do sync always. Problem occur only on “same value assignment”. If “different value assignment”, no problem. e.g.

    test: function(e) { //same value assignment ('update' function will have problem after executed 'test')
        this.localDataSource = JSON.parse(JSON.stringify(this.localDataSource));
    },
    test2: function(e) { //different value assignment ('update' function works fine after executed 'test2')
        let ds = JSON.parse(JSON.stringify(this.localDataSource));
        ds[0]['ProductName'] = 'hello';
        this.localDataSource = JSON.parse(JSON.stringify(ds));
    }

**Comment by posva** [comment]
Hi, thanks for your interest but Github issues are for Vue bug reports and feature requests only. You can ask questions on the [forum](http://forum.vuejs.org/), the [Discord server](https://vue-land.js.org/) or [StackOverflow](http://stackoverflow.com/tags/vue.js).

---

Please reach out to Telerik support on how to use their product

**Comment by icqbox** [comment]
Hi, thanks for your reply.

Before I came here, I posted this question to Telerik Kendo and we both guess it's related to Vue's reactive behaviour.
https://feedback.telerik.com/kendo-vue-ui/1471521-vue-state-dose-not-sync-with-kendo-datasource

At the same time, I also posted to the forum and StackOverflow for a week but am still waiting for reply.
https://forum.vuejs.org/t/vue-js-state-and-kendo-datasource-issue/97689
https://stackoverflow.com/questions/62339856/vue-state-does-not-sync-with-kendo-datasource

Thanks anyway.
